File: app/src/main/java/com/games/memorygame/ui/score/ScorePresenter.java
package com.games.memorygame.ui.score;

import com.games.memorygame.persistence.ScoreDataSource;
import com.games.memorygame.ui.BasePresenter;

import javax.inject.Inject;

import timber.log.Timber;

import static com.games.memorygame.util.RxUtil.call;


public class ScorePresenter extends BasePresenter<ScoreMvpView> {

    private ScoreDataSource scoreLocalDataSource;

    @Inject
    public ScorePresenter(ScoreDataSource scoreDataSource) {
        this.scoreLocalDataSource = scoreDataSource;
    }

    public void loadScores() {
        assertViewAttached();
        addDisposable(call(scoreLocalDataSource.loadScores()).subscribe(
                scores -> getView().displayScores(scores),
                throwable -> {
                    Timber.e(throwable, throwable.getMessage());
                    getView().onError();
                })
        );
    }

}

File: app/src/main/java/com/games/memorygame/ui/score/ScorePresenter.kt
package com.games.memorygame.ui.score

import com.games.memorygame.persistence.ScoreDataSource
import com.games.memorygame.ui.BasePresenter

import timber.log.Timber

import com.games.memorygame.util.observeOnMainThread


class ScorePresenter(private val scoreLocalDataSource: ScoreDataSource) :
        BasePresenter<ScoreMvpView>() {

    fun loadScores() {
        assertViewAttached()
        addDisposable(
                scoreLocalDataSource.loadScores()
                        .observeOnMainThread()
                        .subscribe(
                                { scores -> view!!.displayScores(scores) },
                                { throwable ->
                                    Timber.e(throwable.message)
                                    view?.onError()
                                }
                        )
        )
    }

}
